web/src/typing_events.js

```diff
@@ -114,7 +139,12 @@ export function display_notification(event) {
     const key = get_key(event);
     typing_data.add_typist(key, sender_id);
 
-    render_notifications_for_narrow();
+    if (event.message_type === "message_edit") {
+        const message_id = event.message_id;
+        render_message_editing_typing(message_id, true);
+    } else {
+        render_notifications_for_narrow();
+    }
 
     typing_data.kickstart_inbound_timer(
         key,
```

Comment:
I'm not sure where in this logic it would be the case that when you navigate to a message that someone is actively editing, that fact would be visible.

Reply:
The way that this currently works is -- the typing notification event for a message is sent to all the users subscribed to the stream in which the message is present. So irrespective of whether the user is viewing the message or not, the event would add the `. . .` animation  to the `editing_notifications` <div> present.

However, when a user subscribes to a stream just after a message editing in that stream has started, then he may have to wait until next push for the notification.

Reply:
Ok, I get it. I have updated the code to maintain the states of `message_id` being edited in `typing_data`, which is used by `message_list_view` to render the message edit animation, when user navigates to the stream after event is received.

Reply:
OK and how does the message get its `message_container.is_typing` cleared when typing stops? I'm somewhat worried that a background rerender might cause incorrect display, but maybe the theory is that `set_calculated_message_container_variables` gets called every time a message is rerendered? If so this is likely correct.

Reply:
yes, `set_calculated_message_container_variables` is called every re render.

Reply:
This is resolved in the latest update.